docs/source/guides/schema-design.md

```diff
@@ -0,0 +1,290 @@
+}
+
+type ColoringBook implements Book {
+  title: String
+  author: Author
+  colors: [Color]
+}
+
+type Query {
+  schoolBooks: [Book]
+}
+```
+
+In this example, we've used the `Book` interface as the foundation for the `TextBook` and `ColoringBook` types. Then, a `schoolBooks` field simply expresses that it returns a list of books (i.e. `[Book]`).
+
+Implementing the book feed example is now simplified since we've removed the need to worry about what kind of `Book`s will be returned. A query against this schema, which could return _text books_ and _coloring_ books, might look like:
+
+```graphql
+query GetBooks {
+  schoolBooks {
+    title
+    author
+  }
+}
+```
+
+This is really helpful for feeds of common content, user role systems, and more!
+
+Furthermore, if we need to return fields which are only provided by either `TextBook`s or `ColoringBook`s (not both) we can request fragments from the abstract types in the query. Those fragments will be filled in only as appropriate; in the case of the example, only coloring books will be returned with `colors`, and only textbooks will have `classes`:
+
+```graphql
+query GetBooks {
+  schoolBooks {
+    title
+    ... on TextBook {
+      classes {
+        name
+      }
+    }
+    ... on ColoringBook {
+      colors {
+        name
+      }
+    }
+  }
+}
+```
+
+<h2 id="mutations">Designing mutations</h2>
+
+Mutations are one of the core types in GraphQL. Just like you can make a query to fetch information, you can make a mutation to update or change information. Unlike REST, GraphQL mutations actually are executed in two parts: the mutation itself, and a subsequent query, which can return any kind of data that you normally could query for. A mutation definition for updating the age of a `User` could look like this:
+
+```graphql
+type Mutation {
+  updateUserAge(id: ID!, age: Int!): User
+}
+
+type User {
+  id: ID!
+  name: String!
+  age: Int!
+}
+```
+
+With that definition, you could then make the following mutation:
+
+```graphql
+mutation updateMyUser {
+  updateUserAge(id: 1, age: 25){
+    id
+    age
+    name
+  }
+}
+```
+
+With a response looking something like:
+
+```json
+{
+  "data": {
+    "updateUserAge": {
+      "id": "1",
+      "age": "25",
+      "name": "jane doe"
+    }
+  }
+}
+```
+
+The first thing to note is that it’s most common to return the thing you’re updating from a mutation. In our example, we were updating a `User` record, so we returned that updated user. There’s nothing _enforcing_ this practice, but it’s highly recommended, because it’s often needed to have an updated user to let the clients update any local cache with a new instance of that `User`. For this same reason, it’s useful design mutations to update only one entity. If you wanted to update two unrelated entities, it’s recommended to in separate mutations. For more details on how to handle errors and warnings in mutations, see the [mutation responses](#mutation-responses) section below.
```

Comment:
> The first thing to note is that it’s most common to return the thing you’re updating from a mutation. In our example, we were updating a `User` record, so we returned that updated user.

I think a mutation should frequently return a special return type, like we detail below. And if the mutation does indeed represent an action that updates multiple types of things, it should probably return all of them. This sentence makes it seem like all mutations should just return a single object type, which I don't think is good advice.

> For this same reason, it’s useful design mutations to update only one entity. If you wanted to update two unrelated entities, it’s recommended to in separate mutations.

I actually feel the opposite of this - IMO a mutation should represent a user action on the frontend, regardless of how many entities that updates.